File: python/13.py
class Solution:
    def romanToInt(self, s: str) -> int:
        int_roman_dict: dict[str, int] = {"I": 1, "V": 5, "X": 10, "L": 50, "C": 100, "D": 500, "M": 1000}

        result_sum: int = 0
        prev_val: int = 0

        for roman in s[::-1]:
            current_value: int = int_roman_dict[roman]
            if prev_val > current_value:
                result_sum -= current_value
            else:
                result_sum += current_value
            prev_val = current_value

        return result_sum

File: c/13.c
#include <string.h>

int romanToInt(char *s)
{
  char roman[7] = {'I', 'V', 'X', 'L', 'C', 'D', 'M'};
  int val[7] = {1, 5, 10, 50, 100, 500, 1000};

  int result_sum = 0;
  int previous_val = 0;

  for (int i = strlen(s) - 1; i >= 0; i--)
  {
    int current_value = 0;

    for (int j = 0; j < 7; j++)
    {
      if (roman[j] == s[i])
      {
        current_value = val[j];
        break;
      }
    }

    if (previous_val > current_value)
    {
      result_sum -= current_value;
    }
    else
    {
      result_sum += current_value;
    }
    previous_val = current_value;
  }

  return result_sum;
}
